internal/engine/filter/convert.go

```diff
@@ -227,14 +225,12 @@ func joinQueries(f1, f2 string) string {
 }
 
 func joinStrings(escape bool, a ...string) (string, error) {
-	if escape {
-    // If the strings are already quoted, we would double quote here.
-    // This is not recognized correctly by Gmail and we should fail.
-		for _, a := range a {
-			if strings.Contains(a, `"`) {
-				return "", fmt.Errorf("invalid quote in %q.", a)
-			}
+	for _, a := range a {
```

Comment:
The check has to go within `if escape {`. You want to fail only if the string contains quotes and we also need to add them (escape == true), because this would result in double quoting.

There are cases where strings are already quoted, e.g. when a filter is marked as `isRaw = true` and we don't want to break that use case.

Reply:
Thanks for the catch - I was moving it around when the test was failing and failed to put it back inside